### PR TITLE
Problem: infinite failed to fetch block error might occur on prune node

### LINKED ISSUE #397: Problem: edge case of indexer causes it won't work
Currently, indexer if not able to fetch a block result, it will retry infinity so on a prune-node, there is an edge case that indexer is turned-off at the past, the data on db is exists, LastIndexedBlock returns positive number but that block is already pruned and the indexer can not back to work.

@yihuang can you confirm?

If that is correct, shall we force the `lastBlock` variable to node's earliest/latest height?
The missing gap is not important because the indexer in any node prune will be missing txs the same as it is.

```golang
if lastBlock == -1 {
	lastBlock = latestBlock
} else if lastBlock < status.SyncInfo.EarliestBlockHeight {
	lastBlock = status.SyncInfo.EarliestBlockHeight // or latestBlock
	// kinda unsafe, but we don't have a better way to do this
}
```

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: https://github.com/crypto-org-chain/ethermint/issues/397

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (rpc) [#388](https://github.com/crypto-org-chain/ethermint/pull/388) Avoid out of bound panic when error message.
 * (rpc) [#391](https://github.com/crypto-org-chain/ethermint/pull/391) Align block param with go-ethereum in debug_traceCall.
 - (evm) [#396](https://github.com/crypto-org-chain/ethermint/pull/396) Align evm tx type with go-ethereum.
+* (rpc) [#](https://github.com/crypto-org-chain/ethermint/pull/) Avoid infinite failed to fetch block error when lastBlock is smaller than earliest on prune node.
```
Comment (mmsqe) [inline CHANGELOG.md]:
```suggestion
* (rpc) [#398](https://github.com/crypto-org-chain/ethermint/pull/398) Avoid infinite failed to fetch block error when lastBlock is smaller than earliest on prune node.
```

File: server/indexer_service.go [hunk 2]
```diff
@@ -94,6 +94,13 @@ func (eis *EVMIndexerService) OnStart() error {
 	}
 	if lastBlock == -1 {
 		lastBlock = latestBlock
+	} else if lastBlock < status.SyncInfo.EarliestBlockHeight {
+		// to avoid infinite failed to fetch block error when lastBlock is smaller than earliest
+		lastBlock = status.SyncInfo.EarliestBlockHeight
```
Comment (yihuang) [inline server/indexer_service.go:110]: Does it work? What is the value of earliest block if a node prune in the middle.

Comment (mmsqe) [reply server/indexer_service.go:110]: we might need skip the pruned blocks with retries

Comment (mmsqe) [reply server/indexer_service.go:110]: earliest could be middle block, I could reproduce by turn indexer on and off with small max_age_num_blocks and config like
```
"min-retain-blocks": 10,
"pruning-interval": 10,
"pruning-keep-recent": 4,
"state-sync": {
    "snapshot-interval": 1
}
```

**Review by yihuang — APPROVED** [review]

**Pull request merged**